src/main/java/org/stellar/sdk/xdr/Asset.java

```diff
@@ -79,44 +79,39 @@ public Asset build() {
   }
 
   public static void encode(XdrDataOutputStream stream, Asset encodedAsset) throws IOException {
-    //Xdrgen::AST::Identifier
-    //AssetType
-    stream.writeInt(encodedAsset.getDiscriminant().getValue());
-    switch (encodedAsset.getDiscriminant()) {
-      case ASSET_TYPE_NATIVE:
-        break;
-      case ASSET_TYPE_CREDIT_ALPHANUM4:
-        AlphaNum4.encode(stream, encodedAsset.alphaNum4);
-        break;
-      case ASSET_TYPE_CREDIT_ALPHANUM12:
-        AlphaNum12.encode(stream, encodedAsset.alphaNum12);
-        break;
-      case ASSET_TYPE_POOL_SHARE:
-        throw new RuntimeException("Invalid asset type");
-    }
+  //Xdrgen::AST::Identifier
```

Comment:
any reason why the formatting here got weird? :eyes:

Reply:
none that I can see, to be sure, I recopied latest .x from p19 branch and then regen'd code `xdrgen -o ./src/main/java/org/stellar/sdk/xdr -l java -n org.stellar.sdk.xdr ./xdr/Stellar-types.x ./xdr/Stellar-SCP.x ./xdr/Stellar-overlay.x ./xdr/Stellar-ledger-entries.x ./xdr/Stellar-ledger.x ./xdr/Stellar-transaction.x`, this was the result.